pkg/util/yaml/decoder.go

```diff
@@ -246,16 +246,28 @@ func (r *YAMLReader) Read() ([]byte, error) {
 			return nil, err
 		}
 
-		if string(line) == separator || err == io.EOF {
+		sep := len([]byte(separator))
+		if i := bytes.Index(line, []byte(separator)); i == 0 {
```

Comment:
nit: this should be HasPrefix since this is moderately performance sensitive.  Can you do a small follow up (or open an issue)?